src/utils/viewer/viewer.cpp

```diff
@@ -718,7 +727,13 @@ int Viewer::addObject(const std::string& objectAttrHandle) {
 int Viewer::addTemplateObject() {
   int numObjTemplates = objectAttrManager_->getNumFileTemplateObjects();
   if (numObjTemplates > 0) {
-    return addObject(objectAttrManager_->getRandomFileTemplateHandle());
+    std::string fileTemplateHandle =
+        objectAttrManager_->getRandomFileTemplateHandle();
+    // Get object name to display in logging text
+    std::string fileName =
+        fileTemplateHandle.substr(fileTemplateHandle.find_last_of("/") + 1);
+    lastAddedObjectName_ = fileName.substr(0, fileName.find_first_of('.'));
```

Comment:
The functionality of the two lines above can be replaced by using :
```
lastAddedObjectName_ = Cr::Utility::Directory::splitExtension(
          Cr::Utility::Directory::filename(fileTemplateHandle))
          .first;

```

Reply:
Thanks!